File: Examples/Deprecated/Search-Inventory/Perl/searchDevice.pl
use LWP::UserAgent;
use HTTP::Cookies;
use JSON::RPC::Client2;
use Data::Dumper;

### Initial authentication
my $cookie_jar = HTTP::Cookies->new( ignore_discard => 1 );
my $ua = LWP::UserAgent->new(
    cookie_jar => $cookie_jar,
    ssl_opts   => { SSL_use_cert => 0, verify_hostname => 0 }
);

### Create a JSON-RPC proxy for the inventory service
###
my $netld = JSON::RPC::Client2->new("https://localhost/rest?j_username=admin&j_password=password", $ua);

### Search the inventory
print "Enter an individual IP address or IP/CIDR (eg. 10.0.0.0/24): ";
my $query = <STDIN>;
chomp($query);

my $pageData = $netld->call("Inventory.search", {
        network  => 'Default',
        scheme   => 'ipAddress',
        query    => $query,
        pageData => {
            offset   => 0,
            pageSize => 500,
        },
        sortColumn => 'ipAddress',
        descending => 0
    }
);

print "Search found " . $pageData->{total} . " devices.\n";
print "----------------------------------------------\n";
foreach $device (@{$pageData->{devices}})
{
    print $device->{ipAddress} . "\t" . $device->{hostname} . "\n";
}

### Logout using the security service to be nice to the server
###

$netld->call("Security.logoutCurrentUser");

File: Examples/Deprecated/Search-Inventory/Ruby/searchDevice.rb
#!/usr/bin/env ruby -W0

require 'jimson'

OpenSSL::SSL::VERIFY_PEER = OpenSSL::SSL::VERIFY_NONE

### Create a JSON-RPC proxy for the inventory service
###
netld = Jimson::Client.new("https://10.0.40.50/rest?j_username=admin&j_password=password")

### Search the inventory
print "Enter an individual IP address or IP/CIDR (eg. 10.0.0.0/24): ";
query = gets.strip

pageData = { 'offset' => 0, 'pageSize' => 500 }

pageData = netld['Inventory.search', 'Default', 'ipAddress', query, pageData, 'ipAddress', false]

print "Search found " + pageData['total'].to_s() + " devices.\n"
print "----------------------------------------------\n"

for device in pageData['devices']
   print device['ipAddress'] + "\t" + device['hostname'] + "\n";
end

### Logout using the security service to be nice to the server
###
netld['Security.logoutCurrentUser']
